Libplanet/Net/Protocols/KBucket.cs

```diff
@@ -21,49 +21,45 @@ public KBucket(int size, Random random, ILogger logger)
             _size = size;
             _random = random;
             _logger = logger;
-            _peers = new ConcurrentDictionary<BoundPeer, DateTimeOffset>();
+            _peerStates = new ConcurrentDictionary<Address, PeerState>();
             ReplacementCache = new ConcurrentDictionary<BoundPeer, DateTimeOffset>();
 
             _lastUpdated = DateTimeOffset.UtcNow;
         }
 
-        public int Count => _peers.Count;
+        public int Count => _peerStates.Count;
 
         /// <summary>
         /// Most recently used peer.
         /// </summary>
-        public KeyValuePair<BoundPeer, DateTimeOffset> Head
+        public PeerState Head
         {
             get
             {
-                return _peers.Aggregate(
-                    new KeyValuePair<BoundPeer, DateTimeOffset>(
-                        null,
-                        DateTimeOffset.MinValue
-                    ),
-                    (l, r) => l.Value > r.Value ? l : r
-                );
+                return _peerStates.Aggregate(
+                    (x, y) =>
+                        x.Value.LastUpdated > y.Value.LastUpdated ? x : y
+                ).Value;
             }
         }
 
         /// <summary>
         /// Least recently used peer.
         /// </summary>
-        public KeyValuePair<BoundPeer, DateTimeOffset> Tail
+        public PeerState Tail
         {
             get
             {
-                return _peers.Aggregate(
-                    new KeyValuePair<BoundPeer, DateTimeOffset>(
-                        null,
-                        DateTimeOffset.MaxValue
-                    ),
-                    (l, r) => l.Value < r.Value ? l : r
-                );
+                return _peerStates.Aggregate(
```

Comment:
How about Aggregate from _peerStates.Value instead of _peerStates? it seems that we don't need keys of _peerStates at this time.